docs/misc/resource-routes.md

```diff
@@ -0,0 +1,212 @@
+
+```tsx filename=app/routes/reports.$id[.pdf].tsx
+export async function loader({
+  params,
+}: LoaderFunctionArgs) {
+  const report = await getReport(params.id);
+  const pdf = await generateReportPDF(report);
+  return new Response(pdf, {
+    status: 200,
+    headers: {
+      "Content-Type": "application/pdf",
+    },
+  });
+}
+```
+
+<docs-info>Note the filename uses [escaping patterns][escaping] to include the extension in the URL.</docs-info>
+
+When the user clicks the link from the UI route, they will navigate to the PDF.
+
+## Linking to Resource Routes
+
+<docs-error>It’s imperative that you use <code>reloadDocument</code> on any Links to Resource Routes</docs-error>
+
+There's a subtle detail to be aware of when linking to resource routes. You need to link to it with `<Link reloadDocument>` or a plain `<a href>`. If you link to it with a normal `<Link to="pdf">` without `reloadDocument`, then the resource route will be treated as a UI route. React Router will try to get the data with `fetch` and render the component. Don't sweat it too much, you'll get a helpful error message if you make this mistake.
+
+## Handling different request methods
+
+To handle `GET` requests export a loader function:
+
+```tsx
+import type * as Route from "./+types.resource";
+
+export const loader = async ({
+  request,
+}: Route.LoaderArgs) => {
+  // handle "GET" request
+
+  return {success: true}
+};
+```
+
+To handle `POST`, `PUT`, `PATCH` or `DELETE` requests export an action function:
+
+```tsx
+import type * as Route from "./+types.resource";
+
+export const action = async ({
+  request,
+}: Route.ActionArgs) => {
+  switch (request.method) {
+    case "POST": {
+      /* handle "POST" */
+    }
+    case "PUT": {
+      /* handle "PUT" */
+    }
+    case "PATCH": {
+      /* handle "PATCH" */
+    }
+    case "DELETE": {
+      /* handle "DELETE" */
+    }
+  }
+};
+```
+
+Resource Routes do not support [non-standard HTTP methods][nonstandard-http-methods] - those should be handled by the HTTP server that serves your React Router requests.
+
+## Turbo Stream
+
+Returning bare objects from a Resource Route will automatically encode the responses as a [turbo-stream][turbo-stream], which automatically serializes Dates, Promises, and other objects over the network. React Router automatically deserializes turbo-stream data when you call resource routes from React Router APIs such as `<Form>`, `useSubmit`, or `useFetcher`.
+
+You can use the turbo-stream package to decode a bare turbo-stream Response.
+
+```ts
+import {decode} from "turbo-stream";
+
+const resourceResponse = await fetch("/resource");
+
+const decoded = await decode(resourceResponse.body);
+
+const time = decoded.value.time;
+
+// Await a Promise from the resource route response
+const streamedData = await decoded.value.promiseData;
+
+// Wait for the stream to finish
+await decoded.done;
```

Comment:
It's unfortunately a little more complicated than that, and due to single fetch you get all each route's data back when doing a GET request 🤔 

This makes me kind of second guess whether or not we should encourage/document this pattern @brophdawg11, unless I'm missing something. Or maybe we should just focus specifically on non-GET requests and clarify what happens when you do a GET request

```suggestion
import {decode} from "turbo-stream";

const resourceResponse = await fetch("/resource.data", {method: "POST"});

const decoded = await decode(resourceResponse.body);

const time = decoded.value["routes/resource"].data.time

// Await a Promise from the resource route response
const streamedData = await decoded.value.promiseData;

// Wait for the stream to finish
await decoded.done;
```

Reply:
Sorry, didn't mean to leave such a big diff, this was when I originally thought this was for a `loader`/GET request

Reply:
Yep, if you fetch a resource route with `.data` you get _everything_

[demo](https://stackblitz.com/edit/vitejs-vite-tyjyrf?file=app%2Froutes.ts,app%2Froutes%2Fhome.tsx,app%2Froutes%2Fresource.tsx,app%2Froot.tsx)

Reply:
I've removed it and added a warning, explaining that they should use turbo-stream directly if they want to consume it outside of React Router.